src/app/services/auth.service.ts

```diff
@@ -128,4 +158,42 @@ export class AuthService {
 
     return userRef.set(data);
   }
+
+  //Email-link sign in functions
+  sendSignInLink(email, postSlug) {
+    const actionCodeSettings = {
+      url: `http://localhost:4200/blog/post/${postSlug}`,
+      handleCodeInApp: true
+    };
+    firebase.auth().sendSignInLinkToEmail(email, actionCodeSettings);
+    window.localStorage.setItem('emailForSignIn', email);
+  }
+
+  async confirmSignIn() {
+    if (firebase.auth().isSignInWithEmailLink(window.location.href)) {
+      let email = window.localStorage.getItem('emailForSignIn');
+      if (!email) {
+        email = window.prompt('Please provide your email for confirmation');
+        email = email.toLowerCase();
+      }
+      try{
+        const result = await firebase.auth().signInWithEmailLink(email, window.location.href);
+        window.localStorage.removeItem('emailForSignIn');
+        if(result.user.emailVerified){
+          return [result.user.uid, email];
```

Comment:
It would be easier to read, understand, and maintain this codebase if we used an object here instead of an array. So that for instance in the component we could access something like "userInfo.userId" and "userInfo.email" instead of "userInfo[0]" and "userInfo[1]"